openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -66,25 +64,27 @@ def validate_instance(self, instance: Dict[str, Any]) -> None:
             raise ActionException(
                 "SingleSignOn field attributes are not configured in OpenSlides"
             )
-        additional_required_fields = {
-            value: self.model.saml_id.get_payload_schema()
-            for key, value in self.saml_attr_mapping.items()
-            if key == "saml_id"
-        }
-        additional_optional_fields = {
-            value: cast(Field, getattr(self.model, key, {})).get_payload_schema()
-            for key, value in self.saml_attr_mapping.items()
-            if key != "saml_id" and key in allowed_user_fields
-        }
         self.schema = {
             "$schema": schema_version,
             "title": "create saml account schema",
             "type": "object",
             "properties": {
-                **additional_required_fields,
-                **additional_optional_fields,
+                payload_field: {
+                    "oneOf": [
```

Comment:
`oneOf` will actually lead to such bad error message which you always criticize ;) Why do we need the option to give arrays if we only use the first value anyway?

Also, this will lead to problems once we need the option to fill array fields in the user (e.g. groups).

Reply:
To get rid of the `oneOf`, you could merge the two schemas by using i.e. `type: ["string", "array"]`. But I don't know if the code will get much better by doing this...

Reply:
We had to do, it's customer format